maintenance/projects/js-toolkit/packages/liferay-npm-build-tools-common/src/project/index.ts

```diff
@@ -362,18 +362,55 @@ export class Project {
 		}
 	}
 
-	_loadNpmbundlerrc(): void {
-		const npmbundlerrcPath = this._configFile.asNative;
+	_findAutopresets(): string[] {
+		const {dependencies, devDependencies} = this._pkgJson;
+
+		return Object.keys({
+			...(dependencies || {}),
+			...(devDependencies || {}),
+		}).reduce((autoPresets, pkgName) => {
+			try {
+				const {dependencies} = this.require(pkgName + '/package.json');
+
+				if (dependencies && dependencies['liferay-npm-bundler']) {
+					autoPresets.push(pkgName);
+				}
+			}
+			catch (err) {
 
-		const config = fs.existsSync(npmbundlerrcPath)
-			? readJsonSync(npmbundlerrcPath)
-			: {};
+				// ignore
 
-		// Apply preset if necessary
+			}
 
+			return autoPresets;
+		}, []);
+	}
+
+	_loadDefaultPreset(): string {
 		let presetFilePath;
 
-		if (config.preset === undefined) {
+		const autoPresets = this._findAutopresets();
+
+		if (autoPresets.length > 1) {
+			throw new Error(
+				'Multiple autopreset dependencies found in project (' +
+					autoPresets +
+					'): please remove the invalid ones or ' +
+					'explicitly define the preset to be used in the ' +
+					'.npmbundlerrc file'
```

Comment:
I would soooo love if we could get rid of `.npmbundlerrc`... 🙏

Reply:
What do you mean? We can build projects without `.npmbundlerrc` already, as long as the user is satisfied with the default values. In fact this PR makes it even easier to have projects without `.npmbundlerrc` because you can create an autopreset for that :tada:

Reply:
Before the autopreset only our default preset was "magic", but that was not enough for most of the projects.

Reply:
I wish we didn't have a thing that's `.npmbundlerrc` specially if we build more stuff and then it starts overlapping with `npmscripts.config.js` and eventually others. I just get the feeling there are way too many config marker files all over the place and their names aren't really descriptive/unique to understand what should each of them do.

Reply:
And where would we put that information? I find having a tool like this one (a bundler) without any configuration almost impossible :thinking: 

We also need files like `.babelrc`, `.eslint`, `webpack.config.js`, etc. to build projects. Unless we make something like `npm-scripts` that abstracts you from all that, I can't see the solution (even if we get rid of `.npmbundlerrc`, because we still have the other files).

And, if we make an umbrella tool like `npm-scripts` then, what's the problem with having a `.npmbundlerrc` file created under the hood as we do in portal? It's transparent to the user...

One idea I had long ago was to make a programmatic API for the bundler so that a higher level tool could configure and call it without the need for any file (like you can do with Babel, for example). It shouldn't be that difficult because, since we introduced the `Project` class for abstracting the configuration, the bundler's core is agnostic of the `.npmbundlerrc` (it can come from any other place we like).

Reply:
But even in that case, as I said, we need some other files for eslint, webpack, etc... :-(

Maybe there's some way to internalize them inside the bundler like we can do for babel (you can put babelrc inside npmbundlerrc) but I don't think it's that useful. And it is probably an abuse of the bundler if we want to specialize it in just bundling.

Reply:
> there are way too many config marker files all over the place and their names aren't really descriptive/unique

Do we have a list of what all these would be? Our own homebrewed ones are `.npmbridgerc`,  `.npmbundlerrc` and `npmscripts.config.js` right? Third party ones being eslint, prettier, webpack, babel, 

@jbalsas is your concern due to having the extra file, the name of it, or just the config in general? Or all the above? 

- We could move config into the package.json to avoid an extra file. 
- Naming is hard and especially in a project as large as DXP.
- I don't think it'd be possible to throw out the config altogether, we can provide really good defaults, but there will always be cases that we need to account for that defaults won't satisfy.

Future thought here, would it be possible to converge our configs to be more like `npmscripts.config.js`? If we aim for a singular config files, we might be able to reduce the amount of confusion between each of them. Or at least we could possible aim for a singular config for liferay js related things.

Reply:
> we need some other files for eslint, webpack...

But those are standard, so JS developers or even Java developers expect them. The issue is ours, those are the ones that we need to avoid or make them as descriptive and simple as possible.

Reply:
Yep! Totally agree, @nhpatt.

Reply:
> We could move config into the package.json to avoid an extra file

I don't like this one too much because `package.json` files are published to npmjs.com and there's always the risk of leaking something in the config. I understand that some of the config may be useful for the consumers of the packages, but some other is better left unpublished, I think...

Reply:
Regarding the other files, we have:

- .npmbridgerc
- .npmbundlerrc 
- npmscripts.config.js

Additionally, there's the possibility to embed some other standard files inside .npmbundlerrc. For example, it is possible to embed a .babelrc file inside the .npmbundlerrc (this behaviour is copied from webpack). However, we can also allow to have a standalone .babelrc file next to the .npmbundelerrc if we want. All we need to decide is what we want.

In general, as I said, the bundler uses a .npmbundlerrc file (like any other standard tool, like webpack). That file is fully documented in [our docs](https://github.com/liferay/liferay-frontend-projects/blob/master/maintenance/projects/js-toolkit/docs/.npmbundlerrc-file-reference.md) and it is totally optional because:

1. Every setting has a default
2. If the file is missing, we apply the standard preset automatically (and now, if we merge this, any autopreset that the tools finds configured in the project).

Said that, it's impossible for some projects to have no .npmbundlerrc file because there are some things that are simply impossible to infer (e.g: bundler imports). For that cases, you could create an autopreset (like our target platforms) that configure the bundler without any need to have a .npmbundlerrc.

We also need to consider the **divergence between portal's build and customer's builds**, which are quite different, but the bundler needs to account for both. For example:

- Portal: the JAR is created by gradle, uses src/main/... folders, has npm-scripts on top
- Customers: the JAR is created by blade or the bundler itself, doesn't use src/main/... folders

Also, from the portal's POV, the bundler is something we want to hide, but from the customers' POV the bundler is really a tool. Even if we don't want to expose them to it, there are several customers using it for their own workflows, built on top of it. So, in that regards, it is similar to webpack.

I understand that target platforms are intended to work like npm-scripts (hiding the bundler as much as possible), but we still have the other use case, where advanced users need the bare tool.

----

In any case, ~can we create an issue to track this and decide there exactly what would be ideal for us?~ I've created [an issue](https://github.com/liferay/liferay-frontend-projects/issues/551) to track this. We can add the `liferay-cli` proposal there, and the final layout the different develper roles (portal, customer, advanced customer) would expect.

Regarding this PR, it doesn't change the model of configuration we have now. If any, it enhances it because it goes in the direction of getting rid of the npmbundlerrc.

Reply:
The `.npmbridgerc` file is for a totally different tool that simply creates JS modules that re-export other JS modules. This was created to maintain backwards compatiblity of very old JS modules we had in portal (pre-bundler stuff that used the AMD loader without any versioning of npm packages present in their URLs).

Maybe it is better if we remove all that (I think it could be possible by now) instead of finding a way to hide the `npmbridgerc`. We haven't added more bridges (because they are not really needed) and we wouldn't want to add them :sweat_smile: . 

In the worst case scenario we could always generate the current bridges we have in portal, check them into git and delete the bridge creator tool forever.

This is similar to what we did in the `frontend-js-node-shims` project (checking in hand made custom packages as if they had been built by the bundler, instead of complicating the bundler to build them) and it has gone well: it hasn't given any significant problems, and we are now in a position where we can remove `frontend-js-node-shims` in the future if we want. And all that without having impacted the JS Toolkit.